server/src/index.js

```diff
@@ -0,0 +1,223 @@
+    } catch (error) {
+      console.error('Error loading room messages:', error);
+      connection.send(JSON.stringify({ error: 'Failed to load room messages' }));
+    }
+  })
+
+  connection.on('message', async (message) => {
+    let parsedMessage;
+    try {
+      parsedMessage = JSON.parse(message);
+
+      if (parsedMessage.action === 'joinRoom') {
+        const { roomId, roomName, username } = parsedMessage;
+
+        if (!roomId) {
+          connection.send(JSON.stringify({ error: 'Room ID is required' }));
+          return;
+        }
+
+        const room = await Room.findByPk(roomId);
+        if (!room) {
+          connection.send(JSON.stringify({ error: `Room with ID ${roomId} not found` }));
+          return;
+        }
+
+        connection.roomId = roomId;
+        connection.roomName = roomName;
+        connection.username = username;
+
+        // connection.send(JSON.stringify({ history: room.Messages }));
+        connection.send(JSON.stringify({ success: true, roomId, roomName }));
+
+        wss.clients.forEach(client => {
+          if (client.readyState === WebSocket.OPEN && client.roomId === roomId) {
+            client.send(JSON.stringify({
+              newUser: username,
+              message: `${username} has joined the room.`,
+            }));
+         }
+        });
+      } else if (parsedMessage.action === 'sendMessage') {
+        const { roomId, username, messageText } = parsedMessage;
+
+        if (!roomId || !username || !messageText) {
+          connection.send(JSON.stringify({ error: 'Message or roomId is missing' }));
+          return;
+        }
+
+        const room = await Room.findByPk(roomId);
+        if (!room) {
+          connection.send(JSON.stringify({ error: 'Room not found' }));
+          return;
+        }
+
+        const newMessage = await Message.create({
+          text: messageText,
+          username,
+          RoomId: roomId,
+          // timestamp: new Date().toISOString(),
+          // timestamp: data.timestamp || new Date().toISOString(),
+        });
+
+        wss.clients.forEach(client => {
+          if (client.readyState === WebSocket.OPEN && client.roomId === roomId) {
+            client.send(JSON.stringify({ newMessage: { username, text: messageText, time: new Date().toISOString() } }));
+          }
+        });
+      }
+    } catch (error) {
+      console.error('Error processing WebSocket message:', error);
+      connection.send(JSON.stringify({ error: 'Invalid message format' }));
+    }
+  });
+
+  connection.on('close', () => {
+    connectionsCount--;
+    if (connection.roomId) {
+      wss.clients.forEach(client => {
+        if (client.readyState === WebSocket.OPEN && client.roomId === connection.roomId) {
+          client.send(JSON.stringify({
+            message: `${connection.username} has left the room.`,
+          }));
+        }
+      });
+      console.log('disconected');
+      console.log('disconnected', connectionsCount);
```

Comment:
There is a typo in the console log message 'disconected'. Consider correcting it to 'disconnected' for clarity.